apollo-router/src/query_planner/subgraph_context.rs

```diff
@@ -232,62 +236,65 @@ impl<'a> SubgraphContext<'a> {
 // where we are collecting entites and different entities may have different variables passed to the resolver.
 pub(crate) fn build_operation_with_aliasing(
     subgraph_operation: &SubgraphOperation,
-    contextual_arguments: &Option<ContextualArguments>,
+    contextual_arguments: &ContextualArguments,
     schema: &Schema,
-) -> Result<Operation, ContextBatchingError> {
+) -> Result<Valid<ExecutableDocument>, ContextBatchingError> {
+    dbg!(
+        &subgraph_operation.to_string(),
+        &contextual_arguments,
+        &schema.raw_sdl
+    );
```

Comment:
Does this actually need to return a `ContextBatchingError`? It seems like at the end of the day we either modify or return an operation as is. And when we match on the result in `fetch_node` we return an invalid operation error.

Reply:
I think there's value in knowing the `subgraph_operation` couldn't be parsed, or that due to a bug we couldn't generate a valid operation with aliasing.

That being said I'm not sure what we can do about it, hence the tracing::debug!() for now. but maybe we could add a metric?

as a side note `subgraph_operation.as_parsed(schema.supergraph_schema());` is ... not my favorite structure: https://github.com/apollographql/router/blob/f41065e726b30b52535f7211b5f5ff9db5f4ef20/apollo-router/src/query_planner/fetch.rs#L173

as_parsed takes a schema as parameter, but might not use it if the document has already been parsed (let's hope the schema is the same...)